server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -643,11 +665,19 @@ void checkDeprecation(Settings settings) {
                     key
                 );
             } else {
-                Settings.DeprecationLoggerHolder.deprecationLogger.critical(DeprecationCategory.SETTINGS, key, message, key);
+                Settings.DeprecationLoggerHolder.deprecationLogger.critical(DeprecationCategory.SETTINGS, key, DEPRECATION_MSG, key);
             }
+        } else if (aliasExists(settings)) {
+            logDeprecatedAlias();
         }
     }
 
+    void logDeprecatedAlias() {
```

Comment:
Currently I'm always considering an alias to be deprecated. It's really just a mechanism to migrate a setting and shouldn't be used permanently. 
Though, if preferred, this could be triggered by a new property, e.g. `DeprecatedAlias`.